src/test/scala/org/antipathy/scoozie/builder/TransactionBuilderSpec.scala

```diff
@@ -315,53 +333,63 @@ class TransactionBuilderSpec extends FlatSpec with Matchers {
                          |    ]""".stripMargin
 
     val config = Seq(ConfigFactory.parseString(configString).getConfigList("transitions").asScala: _*)
-
     val startNode = TransitionBuilder.build(config, yarnConfig)
     startNode.name should be("start")
     startNode.failureTransition should be(None)
 
-    val decisionNode = startNode.successTransition.get
-    decisionNode.name should be("decisionNode")
-    decisionNode.action.asInstanceOf[Decision].defaultPath.name should be("sparkAction")
-    decisionNode.action.asInstanceOf[Decision].transitionPaths.map(_.name).toSet should be(
-      Set("sparkAction", "hiveAction")
-    )
-
-    decisionNode.successTransition should be(None)
-    decisionNode.failureTransition should be(None)
-
-    val sparkAction =
-      decisionNode.action.asInstanceOf[Decision].transitionPaths.toSet.filter(_.name == "sparkAction").head
-    sparkAction.name should be("sparkAction")
-    sparkAction.successTransition.get.name should be("shellAction")
-    sparkAction.failureTransition.get.name should be("errorEmail")
-
-    val hiveAction =
-      decisionNode.action.asInstanceOf[Decision].transitionPaths.toSet.filter(_.name == "hiveAction").head
-    hiveAction.name should be("hiveAction")
-    hiveAction.successTransition.get.name should be("shellAction")
-    hiveAction.failureTransition.get.name should be("errorEmail")
-
-    val shellAction = sparkAction.successTransition.get
-    shellAction.name should be("shellAction")
-    shellAction.successTransition.get.name should be("end")
-    shellAction.failureTransition.get.name should be("errorEmail")
-
-    val emailAction = hiveAction.failureTransition.get
-    emailAction.name should be("errorEmail")
-    emailAction.successTransition.get.name should be("kill")
-    emailAction.failureTransition.get.name should be("kill")
-
-    val endAction = shellAction.successTransition.get
-    endAction.name should be("end")
-    endAction.successTransition should be(None)
-    endAction.failureTransition should be(None)
-
-    val killAction = emailAction.successTransition.get
-    killAction.name should be("kill")
-    killAction.successTransition should be(None)
-    killAction.failureTransition should be(None)
-
+    startNode.successTransition.isDefined should be(true)
+    startNode.successTransition.foreach { decisionNode =>
+      decisionNode.name should be("decisionNode")
+      decisionNode.action.asInstanceOf[Decision].defaultPath.name should be("sparkAction")
+      decisionNode.action.asInstanceOf[Decision].transitionPaths.map(_.name).toSet should be(
+        Set("sparkAction", "hiveAction")
+      )
+      decisionNode.successTransition should be(None)
+      decisionNode.failureTransition should be(None)
+      val sparkNode = decisionNode.action.asInstanceOf[Decision].transitionPaths.toSet.find(_.name == "sparkAction")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of untyped comparison operator](https://app.codacy.com/app/SimonJPegg/scoozie/pullRequest?prid=3072461)